src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -1234,6 +1236,86 @@ private IEnumerable<TypeInferenceInfo> InferTypeInPropertyDeclaration(PropertyDe
             return CreateResult(typeInfo.Type);
         }
 
+        private IEnumerable<TypeInferenceInfo> InferTypeInExpressionElement(ExpressionElementSyntax expressionElement)
+        {
+            if (expressionElement.Parent is CollectionExpressionSyntax collectionExpression)
+            {
+                var collectionType = SemanticModel.GetTypeInfo(collectionExpression, CancellationToken).ConvertedType;
+
+                // Try to figure out the type based on the type of hte collection itself.
+                if (CollectionExpressionExtensions.IsConstructibleCollectionType(
+                        SemanticModel.Compilation, collectionType, out var elementTypes) &&
+                    elementTypes is [var elementType, ..])
+                {
+                    return [new(elementType)];
+                }
+
+                // If that fails, see if we can figure out from one of our siblings.
```

Comment:
A potential improvement would be to look at multiple siblings if there's a common type base type and maybe infer that. Thinking of the case where we might build an array of a number of different provider types implementing a common interface:

var a = [new ProviderA(), new ProviderB(), GetSpecialProvider()]

probably should infer the interface type.

Don't feel you need to change this prior to merge...or ever.